app/controllers/your-psp/your-psp-tasks.service.test.js

```diff
@@ -0,0 +1,153 @@
+const { expect } = require('chai')
+const { getCurrentCredential } = require('../../utils/credentials')
+const { getStripeTaskList, stripeTaskListIsComplete } = require('./your-psp-tasks.service')
+const gatewayAccountFixtures = require('../../../test/fixtures/gateway-account.fixtures')
+
+describe('Progress of Stripe PSP tasklist page', () => {
+  it('should correctly calculate all conditions being met for Stripe',  () => {
+    const account = gatewayAccountFixtures.validGatewayAccount({
+      gateway_account_credentials: [
+        { state: 'ACTIVE', payment_provider: 'stripe', id: 100 }
+      ]
+    })
+    account.connectorGatewayAccountStripeProgress = {
+      bankAccount: true,
+      director: true,
+      vatNumber: true,
+      companyNumber: true,
+      responsiblePerson: true,
+      organisationDetails: true,
+      governmentEntityDocument: true
+    }
+    const activeCredential = getCurrentCredential(account)
+    const taskList = getStripeTaskList(activeCredential, account)
+
+    expect(taskList.ENTER_BANK_DETAILS.enabled).to.equal(false)
+    expect(taskList.ENTER_RESPONSIBLE_PERSON.enabled).to.equal(false)
+    expect(taskList.ENTER_DIRECTOR.enabled).to.equal(false)
+    expect(taskList.ENTER_VAT_NUMBER.enabled).to.equal(false)
+    expect(taskList.ENTER_COMPANY_NUMBER.enabled).to.equal(false)
+    expect(taskList.CONFIRM_ORGANISATION_DETAILS.enabled).to.equal(false)
+    expect(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.enabled).to.equal(false)
+
+    expect(taskList.ENTER_BANK_DETAILS.completed).to.equal(true)
+    expect(taskList.ENTER_RESPONSIBLE_PERSON.completed).to.equal(true)
+    expect(taskList.ENTER_DIRECTOR.completed).to.equal(true)
+    expect(taskList.ENTER_VAT_NUMBER.completed).to.equal(true)
+    expect(taskList.ENTER_COMPANY_NUMBER.completed).to.equal(true)
+    expect(taskList.CONFIRM_ORGANISATION_DETAILS.completed).to.equal(true)
+    expect(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.completed).to.equal(true)
+
+    expect(stripeTaskListIsComplete(taskList)).to.equal(true) 
+  })
+
+  it('correctly calculates progress required for Stripe tasks when non of the conditions are met', () => {
```

Comment:
Would be good to start test names with `should` should it reads `it should ...`

```suggestion
  it('should calculate Stripe tasks statuses correctly when none of the tasks are completed', () => {
```